mods/ascent/mobs/nymph/nymph_inventory.dm

```diff
@@ -0,0 +1,98 @@
+/mob/living/carbon/alien/ascent_nymph/drop_from_inventory(var/obj/item/I)
+	. = ..()
+	if(I == holding_item)
+		holding_item = null
+	else if(I == hat)
+		hat = null
+		update_icons()
+
+/mob/living/carbon/alien/ascent_nymph/put_in_hands(var/obj/item/W) // No hands. Use mouth.
+	if(can_collect(W))
+		collect(W)
+	else
+		W.forceMove(get_turf(src))
+	return 1
+
+/mob/living/carbon/alien/ascent_nymph/proc/wear_hat(var/obj/item/clothing/head/new_hat)
+	if(hat || !istype(new_hat) || holding_item == new_hat)
+		return FALSE
+	hat = new_hat
+	hat.forceMove(src)
+	hat.equipped(src)
+	hat.screen_loc = ANYMPH_SCREEN_LOC_HAT
+	update_icons()
+	return TRUE
+
+/mob/living/carbon/alien/ascent_nymph/hotkey_drop()
+	if(holding_item || hat)
+		drop_item()
+	else
+		to_chat(usr, SPAN_WARNING("You have nothing to drop."))
+
+/mob/living/carbon/alien/ascent_nymph/proc/can_collect(var/obj/item/collecting)
+	return (!holding_item && \
+		istype(collecting) && \
+		collecting != hat && \
+		collecting.loc != src && \
+		!collecting.anchored && \
+		collecting.simulated && \
+		collecting.w_class <= can_pull_size \
+	)
+
+/mob/living/carbon/alien/ascent_nymph/proc/contains_crystals(var/obj/item/W)
+	for(var/mat in W.matter)
+		if(mat == /decl/material/solid/mineral/sand)
+			. += W.matter[mat]
+		else if(mat == /decl/material/solid/gemstone/crystal)
+			. += W.matter[mat]
+		else if(mat == /decl/material/solid/mineral/quartz)
+			. += W.matter[mat]
+		else if(mat == /decl/material/solid/glass)
+			. += W.matter[mat]
+
+/mob/living/carbon/alien/ascent_nymph/proc/collect(var/obj/item/collecting)
+	collecting.forceMove(src)
+	holding_item = collecting
+	visible_message(SPAN_NOTICE("\The [src] engulfs \the [holding_item]."))
+
+	// This means nymph can hoover up beakers as a kind of impromptu chem disposal
```

Comment:
More diona nymph copypaste - is this intentional?

Reply:
Yes. I thought it was cute.